### PR TITLE
PCHR-2225: Remove check for entitlements.length as 0 for users

### PR DESCRIPTION
**Pull request by dhavalsharma**
## Overview
Remove check for entitlements.length as 0 for staff as it is being handled by the backend. This is done as other [PR](https://github.com/civicrm/civihr/pull/1857) is already approved.

~Updated test case for `when user has no entitlements` as it was not done as part of earlier [PR](https://github.com/civicrm/civihr/pull/1857).~

~The contact id is required for request instance and it is updated again to include the same.~

---

- [X] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js [hunk 1]
```diff
@@ -1095,9 +1095,6 @@ define([
             type_id: { IN: absenceTypesAndIds.ids }
           }, true) // `true` because we want to use the 'future' balance for calculation
           .then(function (entitlements) {
-            if (!entitlements.length) {
-              return $q.reject('User has no entitlements');
-            }
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js]: remove this line

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/leave-request-ctrl_test.js [hunk 2]
```diff
@@ -1254,23 +1254,6 @@
         });
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/leave-request-ctrl_test.js]: `spyEntitlementAPI`should be removed which was introduced in https://github.com/civicrm/civihr/pull/1857/files.

**Comment by deb1990** [comment]
1. Fix conflict,
2. Squash commits to remove "PCHR-2225: Update test case to use contact id for entitlements"
3. Remove strikethrough comments from Overview as its not required.

**Comment by dhavalsharma** [comment]
1. fixed
2. cannot squash as its the first commit of this branch
3. strikethrough will give context as to why this PR was created in first place

**Pull request merged**